src/wallet/coinselection.h

```diff
@@ -79,21 +104,20 @@ struct OutputGroup
     size_t m_descendants{0};
     CAmount effective_value{0};
     CAmount fee{0};
-    CFeeRate m_effective_feerate{0};
     CAmount long_term_fee{0};
-    CFeeRate m_long_term_feerate{0};
+    CoinSelectionParams m_cs_params;
```

Comment:
hm, not sure how I feel about having a `CoinSelectionParams` in every single `OutputGroup`... could it just have a `GetSelectionAmount(subtract_fee_outputs)` or something?

Reply:
Currently neither of the actual coin selection functions (which are the ones that call `GetSelectionAmount` are aware of subtract fee from outputs. I decided to go this route of including `CoinSelectionParams` in the `OutputGroup`s because it reduces the number of parameters that we need to pass to both `GroupOutputs` and the `OutputGroup` constructor. Since the `CoinSelectionParams` contain the effective feerate and the long term feerate, it also makes it so that the `OutputGroup` does not need to track those feerates separately.

Reply:
But now every single `COutputGroup` has a copy of `CoinSelectionParams` (and afaik this would be the same across all `COuputGroup`s for a coin selection). That's at least an extra `3*sizeof(size_t) + 3*sizeof(CFeeRate) + 2` for every one.

It also seems like an API violation for the groups of UTXOs to have so much extra information about coin selection (e.g. the size of change output spend). I still think callers could provide those parameters in `group.Insert()`, `group.GetSelectionAmount()`, etc.

Reply:
Making it a `const CoinSelectionParams&` would avoid having so many copies.

Of the 8 parameters in `CoinSelectionParams`, it's an even split between the ones relevant to `OutputGroup`s and `GroupOutputs`. 4 of them (`m_effective_feerate`, `m_long_term_feerate`, `m_subtract_fee_outputs`, and `m_avoid_partial_spends`) are used by `GroupOutputs`. I don't think it's that much of a layer violation. Alternatively, we can still pass in `CoinSelectionParams` and just copy out the parameters that we care about?

Reply:
> Alternatively, we can still pass in CoinSelectionParams and just copy out the parameters that we care about?

This sounds like a better option to me

Reply:
Done